### PR TITLE
Port in NBitcoin uint256 span

### PR DESCRIPTION
**Pull request by xandronus**
Use span for uint256 which should improve performance and memory consumption of the full node.

https://github.com/MetacoSA/NBitcoin/commit/8864a47aaf22f11769d65ade609d38d18f90faa7#diff-6b972e431df706b91757f5c8f0bc3229

### CODE REVIEW

File: src/NBitcoin/UInt256.cs [hunk 1]
```diff
@@ -435,11 +506,23 @@ public uint GetLow32()
 
         public override int GetHashCode()
         {
-            return (int)this.pn0;
+            int hash = 17;
```
Comment (dangershony) [inline src/NBitcoin/UInt256.cs]: There was actually a good reason why we just return pn0, for perf its significantly better and pn0 should in theory be as random as hashing all the rest of the value

Comment (xandronus) [reply src/NBitcoin/UInt256.cs]: I've reverted back to pn0 as per the original code

**Comment by dangershony** [comment]
Good I will try this branch once I merge coindb

**Comment by dangershony** [comment]
Running tests in this now with the new coindb branch

**Pull request merged**